src/core/server/saved_objects/service/lib/scoped_client_provider.ts

```diff
@@ -31,8 +31,33 @@
 import { PriorityCollection } from './priority_collection';
 import { SavedObjectsClientContract } from '../../types';
 import { SavedObjectsRepositoryFactory } from '../../saved_objects_service';
-import { ISavedObjectTypeRegistry } from '../../saved_objects_type_registry';
+import {
+  ISavedObjectTypeRegistry,
+  SavedObjectTypeRegistry,
+} from '../../saved_objects_type_registry';
 import { OpenSearchDashboardsRequest } from '../../../http';
+import { ISavedObjectsRepository } from './repository';
+import { OpenSearchClient } from '../../../opensearch';
+import { IOpenSearchDashboardsMigrator } from '../../migrations';
+
+/**
+ * Options passed to each SavedObjectRepositoryFactoryProvider to aid in creating the repository instance.
+ * @public
+ */
+export interface SavedObjectsRepositoryOptions {
+  migrator: IOpenSearchDashboardsMigrator;
+  typeRegistry: SavedObjectTypeRegistry;
+  includedHiddenTypes: string[];
+  client?: OpenSearchClient;
```

Comment:
when is this param needed? If not, can we remove?

Reply:
Good catch. Thanks! I removed as we don't see use case for now for `OpenSearchClient`. If in future, if we come up with use case where `OpenSearchClient` needed, we can add back as optional parameter.